src/Coordination/KeeperServer.cpp

```diff
@@ -789,18 +789,20 @@ std::vector<int64_t> KeeperServer::getDeadSessions()
     return state_machine->getDeadSessions();
 }
 
-bool KeeperServer::applyConfigUpdate(const ClusterUpdateAction & action)
+KeeperServer::ConfigUpdateResult KeeperServer::applyConfigUpdate(
+    const ClusterUpdateAction & action, bool already_accepted)
 {
+    using enum ConfigUpdateResult;
     std::lock_guard _{server_write_mutex};
 
     if (const auto * add = std::get_if<AddRaftServer>(&action))
     {
-        if (raft_instance->get_srv_config(add->id) != nullptr)
-            return true;
+        if (raft_instance->get_srv_config(add->id) != nullptr) return Committed;
```

Comment:
I was thinking about using condition variable and wait for some time. The cv can be notified inside the callback with type `nuraft::cb_func::NewConfig`. It sounds better to me to just move on to the next changes if we don't manage the wait for this specific change in some timeframe.

Reply:
Done, updated. Though I don't like the idea of skipping some servers -- we may end up in an inconsistent state